--- guile-2.0.11/module/oop/goops/save.scm.orig	2013-03-20 21:01:29.081911547 +0200
+++ guile-2.0.11/module/oop/goops/save.scm	2013-03-20 21:01:43.745911860 +0200
@@ -18,8 +18,9 @@
 ;;;; 
 
 
 (define-module (oop goops save)
+  :use-module (srfi srfi-19)
   :use-module (oop goops internal)
   :use-module (oop goops util)
   :re-export (make-unbound unbound?)
   :export (save-objects load-objects restore
@@ -490,8 +491,25 @@
 (define-method (write-readably (o <method>) file env)
   (goops-error "No read-syntax for <method> defined"))
 
 ;;;
+;;; Date and Time
+;;;
+(define-method (enumerate! (self <date>) env) #f)
+
+(define-method (write-readably (self <date>) file env)
+  (display "(julian-day->date " file)
+  (display (date->julian-day self) file)
+  (display ")" file))
+
+(define-method (enumerate! (self <time>) env) #f)
+
+(define-method (write-readably (self <time>) file env)
+  (display "(julian-day->time-utc " file)
+  (display (date->julian-day (time-utc->date self)) file)
+  (display ")" file))
+
+;;;
 ;;; Environments
 ;;;
 
 (define-class <environment> ()
