--- guile-2.0.11/module/Makefile.in~	Wed Apr 10 01:31:59 2013
+++ guile-2.0.11/module/Makefile.in	Wed Apr 10 18:42:50 2013
@@ -1681,9 +1681,9 @@
 	ice-9/threads.scm ice-9/top-repl.scm ice-9/buffered-input.scm \
 	ice-9/time.scm ice-9/history.scm ice-9/channel.scm \
 	ice-9/pretty-print.scm ice-9/ftw.scm ice-9/gap-buffer.scm \
 	ice-9/weak-vector.scm ice-9/list.scm ice-9/serialize.scm \
-	ice-9/local-eval.scm $(am__append_1)
+	ice-9/colorized.scm ice-9/local-eval.scm $(am__append_1)
 SRFI_SOURCES = \
   srfi/srfi-2.scm \
   srfi/srfi-4.scm \
   srfi/srfi-4/gnu.scm \
