--- guile-2.0.11/module/oop/goops/save.scm.orig	2013-03-19 17:45:43.479607020 +0200
+++ guile-2.0.11/module/oop/goops/save.scm	2013-03-19 17:46:21.055606919 +0200
@@ -20,13 +20,16 @@
 
 (define-module (oop goops save)
   :use-module (oop goops internal)
   :use-module (oop goops util)
-  :re-export (make-unbound)
+  :re-export (make-unbound unbound?)
   :export (save-objects load-objects restore
 	   enumerate! enumerate-component!
 	   write-readably write-component write-component-procedure
-	   literal? readable make-readable))
+	   add-patcher!
+	   literal? readable make-readable
+	   excluded? access-for-each))   
+
 
 ;;;
 ;;; save-objects ALIST PORT [EXCLUDED] [USES]
 ;;;
